src/Sulu/Bundle/AdminBundle/Resources/js/services/Requester/Requester.js

```diff
@@ -12,25 +13,84 @@ const defaultOptions = {
 function transformResponseObject(data: Object) {
     return Object.keys(data).reduce((transformedData: Object, key) => {
         const value = data[key];
-        transformedData[key] = value === null ? undefined : data[key];
+
+        if (value === null) {
+            transformedData[key] = undefined;
+
+            return transformedData;
+        }
+
+        if (Array.isArray(value)) {
+            transformedData[key] = transformResponseArray(value);
+
+            return transformedData;
+        }
+
+        if (value instanceof Object) {
+            transformedData[key] = transformResponseObject(value);
+
+            return transformedData;
+        }
+
+        transformedData[key] = value;
+
         return transformedData;
     }, {});
 }
 
 function transformResponseArray(data: Array<Object>) {
-    return data.map(transformResponseObject);
+    return data.map((value) => {
+        if (value instanceof Object) {
+            return transformResponseObject(value);
+        }
+
+        return value;
+    });
 }
 
 function transformRequestObject(data: Object): Object {
     return Object.keys(data).reduce((transformedData: Object, key) => {
-        transformedData[key] = data[key] === undefined ? null : data[key];
+        const value = data[key];
+
+        if (value === undefined || value === null) {
+            transformedData[key] = null;
+
+            return transformedData;
+        }
+
+        if (Array.isArray(value)) {
+            transformedData[key] = transformRequestArray(value);
+
+            return transformedData;
+        }
+
+        if (value instanceof Object) {
+            transformedData[key] = transformRequestObject(value);
+
+            return transformedData;
+        }
+
+        transformedData[key] = value;
+
         return transformedData;
     }, {});
 }
 
-function transformRequestData(data: Object | Array<Object>): Object | Array<Object> {
+function transformRequestArray(data: Array<Object>) {
+    return data.map((value) => {
+        if (value instanceof Object) {
+            return transformRequestObject(value);
+        }
+
+        return value;
+    });
+}
+
+function transformRequestData(originalData: Object | Array<Object>): Object | Array<Object> {
+    const data = toJS(originalData);
```

Comment:
Can you add a test with some observable data for that? That's something that might be easily overseen in the future...

Reply:
is added